app/celery/service_callback_tasks.py

```diff
@@ -80,11 +84,73 @@ def _send_data_to_service_callback_api(self, data, service_callback_url, token,
         current_app.logger.warning(
             f"{function_name} request failed for notification_id: {notification_id} and url: {service_callback_url}. exc: {e}"
         )
+
+        # TODO: Instate once we monitor alarms to determine how often this happens and we implement
+        #       check_cloudwatch_for_callback_failures(), otherwise we risk flooding the service
+        #       owner's inbox with callback failure email notifications.
+
+        # if isinstance(e, Timeout) or isinstance(e, InvalidURL) or e.response.status_code == 500:
+        #     if check_cloudwatch_for_callback_failures():
+        #         send_email_callback_failure_email(current_app.service)
+
         # Retry if the response status code is server-side or 429 (too many requests).
         if not isinstance(e, HTTPError) or e.response.status_code >= 500 or e.response.status_code == 429:
             try:
                 self.retry(queue=QueueNames.CALLBACKS_RETRY)
             except self.MaxRetriesExceededError:
                 current_app.logger.warning(
-                    "Retry: {function_name} has retried the max num of times for callback url {service_callback_url} and notification_id: {notification_id}"
+                    f"Retry: {function_name} has retried the max num of times for callback url {service_callback_url} and notification_id: {notification_id} for service: {current_app.service.id}"
```

Comment:
For the limits feature, we use an expiring redis key to ensure only one email is sent within a particular time period. For example, if they reach the threshold, we send the email then set this key in redis to true. Then if they hit the threshold again within the time period the redis key will still be true and we will check it before sending another email.